4-WebApp-your-API/4-3-AnyOrg/TodoListService/Controllers/TodoListController.cs

```diff
@@ -10,60 +10,115 @@
 using Microsoft.Identity.Web.Resource;
 using Microsoft.Identity.Web;
 using Microsoft.Identity.Client;
-using System.Net.Http.Headers;
 using Microsoft.Graph;
 using System.Net;
+using Microsoft.Extensions.Configuration;
+using System.Security.Claims;
+using ToDoListService.Extensions;
 
 namespace ToDoListService.Controllers
 {
     [Authorize]
     [Route("api/[controller]")]
-    [RequiredScope("access_as_user")]
     [ApiController]
     public class TodoListController : ControllerBase
     {
         private readonly TodoContext _context;
         private ITokenAcquisition _tokenAcquisition;
-     
-        public TodoListController(TodoContext context,ITokenAcquisition tokenAcquisition)
+        private readonly string[] _graphScopes;
+        private readonly MicrosoftIdentityConsentAndConditionalAccessHandler _consentHandler;
+        private readonly GraphServiceClient _graphServiceClient;
+        IHttpContextAccessor _httpContextAccessor;
+        private readonly string _userTenantId;
+        private readonly string _signedInUser;
+
+        private const string _todoListReadScope = "ToDoList.Read";
```

Comment:
Can we read the scopes from `appsettings.json` instead? [e.g.](https://github.com/Azure-Samples/ms-identity-javascript-angular-tutorial/blob/main/5-AccessControl/1-call-api-roles/API/TodoListAPI/appsettings.json#L6)